web/src/pages/Main.tsx

```diff
@@ -32,10 +33,18 @@ const DividerWithLargeSpacings = styled(Divider)`
   margin-top: 40px;
 `
 
+const sortComponents = (components: TComponentProperty[]) => {
+  const firsts = orderedComponents
+    .map((id) => components.find((x) => x.id === id))
+    .filter((x): x is TComponentProperty => !!x)
```

Comment:
No biggie, but if we are going to use find/lookup anyways, couldn't we just pass response.data.components directly which is a dict/object and create the list/TComponentProperty[] inside sortComponents using the custom order?

Reply:
You're right, this was better